bootstrap/090_post_setup.sql

```diff
@@ -70,21 +70,84 @@ CREATE OR REPLACE TASK TASKS.WAREHOUSE_EVENTS_MAINTENANCE
     ALLOW_OVERLAPPING_EXECUTION = FALSE
     USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE = "XSMALL"
     AS
-    CALL INTERNAL.refresh_warehouse_events(true);
+DECLARE
+    start_time timestamp_ltz default (select current_timestamp());
+    task_run_id text default (select INTERNAL.TASK_RUN_ID());
+    task_name text default 'WAREHOUSE_EVENTS_MAINTENANCE';
+    object_name text default 'WAREHOUSE_EVENTS_HISTORY';
+BEGIN
+    let query_id text := (select query_id from table(information_schema.task_history(TASK_NAME => :task_name)) WHERE GRAPH_RUN_GROUP_ID = :task_run_id  AND DATABASE_NAME = current_database() limit 1);
+    let input variant := (select output from INTERNAL.TASK_LOG where success AND task_name = :task_name AND object_name = :object_name order by task_start desc limit 1);
+    INSERT INTO INTERNAL.TASK_LOG(task_start, task_run_id, query_id, input, task_name, object_name) select :start_time, :task_run_id, :query_id, :input, :task_name, :object_name;
+
+    let output variant;
+    CALL INTERNAL.refresh_warehouse_events(true, :input) into :output;
+
+    let success boolean := (select :output['SQLERRM'] is null);
+    UPDATE INTERNAL.TASK_LOG SET success = :success, output = :output, task_finish = current_timestamp() WHERE task_start = :start_time AND task_run_id = :task_run_id;
+END;
 
 CREATE OR REPLACE TASK TASKS.SIMPLE_DATA_EVENTS_MAINTENANCE
     SCHEDULE = '60 minute'
     ALLOW_OVERLAPPING_EXECUTION = FALSE
     USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE = "XSMALL"
     AS
-    CALL INTERNAL.refresh_all_simple_tables(true);
+DECLARE
+    task_name text default 'SIMPLE_DATA_EVENTS_MAINTENANCE';
+    task_run_id text default (select INTERNAL.TASK_RUN_ID());
+BEGIN
+    let query_id text := (select query_id from table(information_schema.task_history(TASK_NAME => :task_name)) WHERE GRAPH_RUN_GROUP_ID = :task_run_id  AND DATABASE_NAME = current_database() limit 1);
```

Comment:
i think you can pass in the task run id here. or maybe the task root id. Maybe that would be safer. (Imagine our systems where we have several databases with the same task name.)